utils/daos_build.py

```diff
@@ -35,18 +35,32 @@ def load_mpi_path(env):
     if mpicc:
         env.PrependENVPath("PATH", os.path.dirname(mpicc))
 
+def clear_icc_env(env):
+    """Remove icc specific options from environment"""
+    if env.subst("$COMPILER") == "icc":
+        linkflags = str(env.get("LINKFLAGS")).split()
+        if '-static-intel' in linkflags:
+            linkflags.remove('-static-intel')
+        for flag_type in ['CCFLAGS', 'CXXFLAGS', 'CFLAGS']:
+            oldflags = str(env.get(flag_type)).split()
+            newflags = []
+            for flag in oldflags:
+                if 'diag-disable' in flag:
```

Comment:
I would probably have used flag.startswith('-daig-') here but I think it would do the same thing.